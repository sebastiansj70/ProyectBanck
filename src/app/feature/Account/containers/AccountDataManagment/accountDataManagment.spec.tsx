import '@testing-library/jest-dom';
import { RenderResult, render } from '@testing-library/react';
import { AccountDataManagment } from './';
import React from 'react';

describe('Data my account', () => {
  let componenWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof AccountDataManagment>;

  beforeEach(() => {
    componentProps = {
      accountLists: {
        cedula: '123456789',
        nombres: 'Juan Sebastian',
        apellidos: 'Sanchez Parada',
        cuentas: [],
      },
      account: {
        cuentaId: '111 111 111 111',
        tipoCuenta: 'Cuenta de ahorros',
        estado: 1,
        valorDisponible: 25000,
        movimientos: [],
      },
      status: 'ok',
      saveStatusLogin: jest.fn(),
    };

    componenWrapper = render(<AccountDataManagment {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componenWrapper.container).toMatchSnapshot();
  });

  it('contain information', () => {
    const element = componenWrapper.container;
    const tittle = element.querySelectorAll('h3');

    expect(tittle[0].textContent).toMatch(/Informaci√≥n/);
  });
});
