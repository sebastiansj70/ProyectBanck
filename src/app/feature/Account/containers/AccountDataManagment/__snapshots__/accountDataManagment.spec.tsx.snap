// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data my account should match snapshot 1`] = `
<div>
  <div
    class="data-account"
  >
    <div
      class="contenet-account"
    >
      <h3
        class="data-account-info"
      >
        Informaci√≥n
      </h3>
      <div
        class="body-data-account"
      >
        <div
          class="body-info-account"
        >
          <div
            class="user-info-account"
          >
            <h5
              class="tittle-data-account"
            >
              Nombre:
            </h5>
            <h5
              class="name-account"
            >
              Juan Sebastian Sanchez Parada
            </h5>
          </div>
          <div
            class="user-info-account"
          >
            <h5
              class="tittle-data-account"
            >
              Cedula:
            </h5>
            <h5
              class="name-account"
            >
              123456789
               
            </h5>
          </div>
          <div
            class="user-info-account"
          >
            <h5
              class="tittle-data-account"
            >
              Cuenta:
            </h5>
            <h5
              class="name-account"
            >
              111 111 111 111
            </h5>
          </div>
          <div
            class="user-info-account"
          >
            <h5
              class="tittle-data-account"
            >
              Tipo de cuenta:
            </h5>
            <h5
              class="name-account"
            >
              Cuenta de ahorros
            </h5>
          </div>
          <div
            class="user-info-account"
          >
            <h5
              class="tittle-data-account"
            >
              Estado:
            </h5>
            <h5
              class="name-account"
            >
              Activo
            </h5>
          </div>
          <div
            class="user-info-account"
          >
            <h5
              class="tittle-data-account"
            >
              Valor disponible:
            </h5>
            <h5
              class="name-account"
            >
              25000
            </h5>
          </div>
        </div>
        <div
          class="body-describe-info"
        >
          <h3>
            Transacciones 
          </h3>
          Cargando
        </div>
      </div>
    </div>
  </div>
</div>
`;
